src/main/java/io/confluent/connect/jdbc/sink/dialect/DbDialect.java

```diff
@@ -154,8 +156,13 @@ static void formatColumnValue(StringBuilder builder, Schema.Type type, Object va
     }
   }
 
-  protected String getSqlType(Schema.Type type) {
-    final String sqlType = schemaTypeToSqlTypeMap.get(type);
+  protected String getSqlType(String schemaName, Schema.Type type) {
+    final String sqlType;
+    if (schemaName != null && logicalNameToSqlTypeMap.containsKey(schemaName)) {
+      sqlType = logicalNameToSqlTypeMap.get(schemaName);
```

Comment:
We can avoid the `containsKey()` lookup and just do the `get()`, and then do a  `schemaTypeToSqlTypeMap` lookup `if (sqlType == null)`